File: aladdinworks3_ui/src/services/PowerSwitchService.js
import http from "../http-common"; 

class PowerSwitchService {
  getAllPowerSwitchs(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/powerSwitch/powerSwitchs`, searchDTO);
  }

  get(powerSwitchId) {
    return this.getRequest(`/powerSwitch/${powerSwitchId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/powerSwitch?field=${matchData}`, null);
  }

  addPowerSwitch(data) {
    return http.post("/powerSwitch/addPowerSwitch", data);
  }

  update(data) {
  	return http.post("/powerSwitch/updatePowerSwitch", data);
  }
  
  uploadImage(data,powerSwitchId) {
  	return http.postForm("/powerSwitch/uploadImage/"+powerSwitchId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new PowerSwitchService();

File: src/main/java/com/aladdinworks3/service/PowerSwitchService.java
package com.aladdinworks3.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.aladdinworks3.domain.PowerSwitch;
import com.aladdinworks3.dto.PowerSwitchDTO;
import com.aladdinworks3.dto.PowerSwitchSearchDTO;
import com.aladdinworks3.dto.PowerSwitchPageDTO;
import com.aladdinworks3.dto.PowerSwitchConvertCriteriaDTO;
import com.aladdinworks3.service.GenericService;
import com.aladdinworks3.dto.common.RequestDTO;
import com.aladdinworks3.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface PowerSwitchService extends GenericService<PowerSwitch, Integer> {

	List<PowerSwitch> findAll();

	ResultDTO addPowerSwitch(PowerSwitchDTO powerSwitchDTO, RequestDTO requestDTO);

	ResultDTO updatePowerSwitch(PowerSwitchDTO powerSwitchDTO, RequestDTO requestDTO);

    Page<PowerSwitch> getAllPowerSwitchs(Pageable pageable);

    Page<PowerSwitch> getAllPowerSwitchs(Specification<PowerSwitch> spec, Pageable pageable);

	ResponseEntity<PowerSwitchPageDTO> getPowerSwitchs(PowerSwitchSearchDTO powerSwitchSearchDTO);
	
	List<PowerSwitchDTO> convertPowerSwitchsToPowerSwitchDTOs(List<PowerSwitch> powerSwitchs, PowerSwitchConvertCriteriaDTO convertCriteria);

	PowerSwitchDTO getPowerSwitchDTOById(Integer powerSwitchId);







}





